File: worker/internal/domain/weather.go
package domain

type WeatherData struct {
	Temperature     float64 `json:"temperature"`
	Humidity        float64 `json:"humidity"`
	WindSpeed       float64 `json:"wind_speed"`
	RainProbability float64 `json:"rain_probability"`
	CollectedAt     string  `json:"collected_at"`
}

File: api/src/domain/weather/entities/weather.ts
import { Entity } from '@/core/entities/entity'

export interface WeatherProps {
  temperature: number
  humidity: number
  windSpeed: number
  rainProbability: number
  collectedAt: string
}

export class Weather extends Entity<WeatherProps> {
  get temperature() {
    return this.props.temperature
  }

  get humidity() {
    return this.props.humidity
  }

  get windSpeed() {
    return this.props.windSpeed
  }

  get rainProbability() {
    return this.props.rainProbability
  }

  get collectedAt() {
    return this.props.collectedAt
  }

  static create(props: WeatherProps, id?: string) {
    return new Weather(props, id)
  }
}
